compiler/parser/RdrHsSyn.hs

```diff
@@ -2585,23 +2627,40 @@ mkTemplateClassInstanceMethods conName ValidTemplateBody{..} =
                           (mkUnqualVar $ mkVarOcc "cid"))
                         (mkQualVar $ mkDataOcc "Archive")
 
+templateConstraintsToContext :: [Located TemplateConstraint] -> LHsContext GhcPs
```

Comment:
Hopefully we get rid of this if we can change the parse rule `template_hdr` as I suggested.

Reply:
Hopefully - can you see how to salvage https://github.com/digital-asset/ghc/pull/32/commits/5dc09d9d072f5be9d7b887e6b481994110aa7885 from a glance?